### ISSUE TITLE
[BUG] "import cudf" has changed the device ID

### ISSUE THREAD

**Issue by wbo4958** [rapidsai/cudf#11386]
Looks like "import cudf" will change the device ID on 22.06 release

## rapids 22.06

``` bash
conda create -n rapids-22.06 -c rapidsai -c nvidia -c conda-forge  \
    rapids=22.06 python=3.9 cudatoolkit=11.5
```

## repro

``` python
In [1]: import cupy

In [2]: cupy.cuda.runtime.getDevice()
Out[2]: 0

In [3]: cupy.cuda.runtime.setDevice(1)

In [4]: cupy.cuda.runtime.getDevice()
Out[4]: 1

In [5]: import cudf

In [6]: cupy.cuda.runtime.getDevice()
Out[6]: 0

```

**Comment by shwina** [comment]
One way to work around this bug is to import `cudf` before `cupy`. 

If for some reason you cannot do that, yet another workaround involves calling `cuda.cudart.cudaGetDevice()` before importing `cudf`:

```python

In [1]: import cupy

In [2]: import cuda.cudart

In [3]: cuda.cudart.cudaGetDevice()
Out[3]: (<cudaError_t.cudaSuccess: 0>, 0)

In [4]: cupy.cuda.runtime.getDevice()
Out[4]: 0

In [5]: cupy.cuda.runtime.setDevice(1)

In [6]: cupy.cuda.runtime.getDevice()
Out[6]: 1

In [7]: import cudf

In [8]: cupy.cuda.runtime.getDevice()
Out[8]: 1


```

**Comment by wbo4958** [comment]
Thx @shwina

**Comment by vyasr** [comment]
@shwina is this something we need to follow up with the cuda-python team on?

**Comment by wence-** [comment]
This is fixed by the changes in cuda-python that fixed nvidia/cuda-python#24, which are in v11.8. So should be resolved once we move there.

**Comment by vyasr** [comment]
Verified that this no longer occurs:
```

In [1]: import cupy

In [2]: cupy.cuda.runtime.getDevice()
Out[2]: 0

In [3]: cupy.cuda.runtime.setDevice(1)

In [4]: cupy.cuda.runtime.getDevice()
Out[4]: 1

In [5]: import cudf

In [6]: cupy.cuda.runtime.getDevice()
Out[6]: 1
```
